manager/orchestrator/global/global_test.go

```diff
@@ -470,7 +479,8 @@ func updateService(t *testing.T, s *store.MemoryStore, service *api.Service, for
 
 func deleteService(t *testing.T, s *store.MemoryStore, service *api.Service) {
 	s.Update(func(tx store.Tx) error {
-		assert.NoError(t, store.DeleteService(tx, service.ID))
+		service.MarkedForRemoval = true
```

Comment:
Would it make sense to do a `store.GetService`, then update the service, as is done for `updateService`, and not setting the `MarkedForRemoval` flag to false everywhere?  

That way you don't accidentally set the `MarkedForRemoval` flag on the original service object.  I'm mainly a little wary of blanket-setting the `MarkedForRemoval` flag to false in all these tests, because that may cover up a potential bug where the field should not have been set.